components/requests/modals/AssignNumberModal.tsx

```diff
@@ -0,0 +1,87 @@
+import {
+  useDisclosure,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  FormControl,
+  FormLabel,
+  FormErrorMessage,
+  NumberInput,
+  NumberInputField,
+  Button,
+  Text,
+  Box,
+} from '@chakra-ui/react'; // Chakra UI
+import { useState, ReactNode } from 'react'; // React
+
+type AssignNumberModalProps = {
+  readonly onAssign: (num: number) => void;
+  readonly modalTitle: string;
+  readonly fieldName: string;
+  readonly children: ReactNode;
+};
+
+export default function AssignNumberModal({
```

Comment:
It's not so clear what this component does purely from its name. Could you please add a JSDoc comment for this component?